### PR TITLE
Replace SaveWithMesh with separate Save functions

### PR DESCRIPTION
**Pull request by pazner**
(Alternative saving interface for #942)

What do you think about this interface for saving GridFunctions and Meshes @jakubcerveny @tzanio?

### CODE REVIEW

**Comment by pazner** [comment]
@adrienbernede, are the GitHub Actions supposed to be run on PRs whose base branch is not master? I don't see them here.

**Comment by tzanio** [comment]
Either option is fine with me :)

**Comment by tzanio** [comment]
> @adrienbernede, are the GitHub Actions supposed to be run on PRs whose base branch is not master? I don't see them here.

Based on https://github.com/mfem/mfem/tree/example0-save-gf-mesh/.github/workflows I believe the GH actions are run only for PRs toward `master`

**Comment by pazner** [comment]
It might make sense to change the workflows so that they are run on all PRs.

File: fem/gridfunc.hpp [hunk 1]
```diff
@@ -671,10 +671,9 @@ class GridFunction : public Vector
    /// Save the GridFunction to an output stream.
    virtual void Save(std::ostream &out) const;
```
Comment (jakubcerveny) [inline fem/gridfunc.hpp:672]: We could add a `Print` method to mirror the `Mesh` interface, and make `Save(std::ostream&)` call it. I'm just not sure it's not too much :-)

**Review by jakubcerveny — APPROVED** [review]
I like this, thanks!

File: fem/gridfunc.hpp (hunk 1 above)
Comment (tzanio) [reply fem/gridfunc.hpp:672]: I'd prefer to keep it simple for the users and unify to `Save`.

The Save/Print has been pointed out as one of the interface inconsistencies in https://github.com/mfem/mfem/issues/1557

**Review by tzanio — APPROVED** [review]

**Review by barker29 — APPROVED** [review]
I think this is slightly better than what's currently in #942 but do not feel strongly about it, either way is fine.

**Pull request merged**